lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1577,13 +1577,19 @@ def cmd_set(*args)
     name  = args[0]
     value = args[1, args.length-1].join(' ')
 
-    # Set PAYLOAD by index
+    # Set PAYLOAD
     if name.upcase == 'PAYLOAD' && active_module && (active_module.exploit? || active_module.evasion?)
-      index_from_list(payload_show_results, value) do |mod|
-        return false unless mod && mod.respond_to?(:first)
+      if value.match(/^(\/?payload\/)|^\//i)
```

Comment:
Can we get some consistency with https://github.com/rapid7/metasploit-framework/pull/12946#issuecomment-587517929? There are cases that I would like shared between the two codes.

Reply:
Also, it may be preferable to [use %r](https://github.com/rubocop-hq/ruby-style-guide#r).

Reply:
The `start_with?` and `end_with?` checks can probably be ignored, but they were done for extra insurance against bad replacements and negligible performance improvement (this code isn't called during Framework load).

I'm mostly concerned with addressing the common cases consistently. I reverted accommodating typos in #8882, but if you'd like to support that, both implementations should!

Reply:
Changed code to be more consistent in style with https://github.com/rapid7/metasploit-framework/pull/12946#issuecomment-587517929.

Reply:
Did you push this change?

Reply:
Sorry, forgot to push it up before I went home for the day :/

Reply:
No worries!

Reply:
Due to the different pathing for payloads, it does not make sense to check for `modules/`, so this looks great.